### ISSUE TITLE
Codi no obert

### ISSUE THREAD

**Issue by adriaaah** [s3rrallonga/s3rrallonga.github.io#4]
Us heu plantejat obrir i publicar el codi?

**Comment by s3rrallonga** [comment]
Hola. 

I tant que ens ho hem plantejat! i no dubtis que és la nostra fita. El nostre objectiu és que aquesta APP pugui ser  utilitzada a nivell global per qualsevol grup de persones que vulguin auto organitzar-se sense patir censura. Entenem que això passa per publicar el codi sota una llicència tipus copyleft.

No obstant necessitem una mica de temps per fer-ho, és molt nova i cal revisar varis aspectes importants. No podem publicar el codi fins que no estem 100% segurs de que no hi ha cap persona ni res en risc.

Esperem que la comunitat open-source ho entengui :)

**Comment by Ryozuki** [comment]
Si la app esta programada en java, es facil de fer un reverse engineer i si lha heu obfuscat, igualment no espereu gaire seguretat, mes haviat temps.

Dons, no entenc perque no es open source directament, mes quan es una app decentralitzada oi?

**Comment by adriaaah** [comment]
Bé, gràcies per la resposta.
Personalment, no em preocupa la comunitat open-source sinó la free-software (programari lliure).

Si no us fa res, deixaré la "issue" oberta fins que el codi sigui publicat.

**Comment by DavidBernal** [comment]
@titansmc suposo que es p2p. Si fos jo el responsable, haguès triat https://dat.foundation/

**Comment by IsmaelMartinez** [comment]
El codi esta semi disponible. Qualsevol que hagui jugat una mica amb Android pot tenir el codi (pero he mirat una mica aqui i alli i no sembla haber res serios). Potser tindrieu que signar el codi (per que trigin una mica mes en descifrar el que hi ha al darrera). @DavidBernal, sembla que han tirat per (retroshare)[https://github.com/RetroShare/RetroShare]. Sembla tan bona alternativa com la que tu aportes.

**Comment by sim6** [comment]
Així s'està violant la GPL?

**Comment by sergivb01** [comment]
> Així s'està violant la GPL?

Sí, el codi hauria de ser obert utilitzant GPL també, pel que tinc entès.

**Comment by IsmaelMartinez** [comment]
@sim6, no hi ha llicència associada al projecte així que no. El que jo personalment posaria una llicència encara que fos MIT. Encara que tinguessis la llicència associada, no hi ha res que digui que el codi ha de ser "open source". És una mica estranya però millor un amb més estudis en el tema pot aportar una resposta més acurada.

**Comment by sergivb01** [comment]
Retroshare, que és el que utilitza la aplicació, [sí té una llicència](https://github.com/RetroShare/RetroShare/blob/master/LICENSE.txt).

**Comment by IsmaelMartinez** [comment]
Aquesta pregunta comença a ser molt interessant. 

Tècnicament, pel que vaig veure, utilitza retroshare com una llibreria (potser fent-ho com un plugin pero no he mirat els detalls).

No és, per se, una modificacio de retroshare (pel que tinc entès que no és massa perquè no puc veure el codi). 

Per això, no sé si tan sols seria necessari llistà el que s'està utilitzant d'open source o s'aplicaria la llicència.

Això és el que s'estan saltant totes les empreses, però seria interessant veure el que un advocat especialista té que dir... Algú a la sala?

**Comment by retroixer** [comment]
[Retroshare és GPL](https://github.com/RetroShare/RetroShare/blob/master/LICENSE.txt)


[Discussió sobre llibreries GPL](https://en.wikipedia.org/wiki/GNU_General_Public_License#Libraries)

> However, if one releases a GPL-licensed entity to the public, there is an issue regarding linking: namely, whether or not a proprietary program that uses a GPL library is in violation of the GPL.
> 
> This key dispute is whether non-GPL software can legally statically link or dynamically link to GPL libraries. Different opinions exist on this issue. The GPL is clear in requiring that all derivative works of code under the GPL must themselves be under the GPL. Ambiguity arises with regards to using GPL libraries, and bundling GPL software into a larger package (perhaps mixed into a binary via static linking). This is ultimately a question not of the GPL per se, but of how copyright law defines derivative works.

**Comment by alexandresaiz** [comment]
Feu-la open-source just l’endema de declarar la independencia. Els geeks viurem igual sense ella i vosaltres us posarieu massa en l’ull de l’huracà...
